### PR TITLE
Eliminating a player #95

### PR DESCRIPTION
**Pull request by IvanovAndrew**

### CODE REVIEW

File: Bang/Players/Player.cs [hunk 1]
```diff
@@ -145,7 +150,10 @@ public void RegainLifePoint()
             PlayerTablet.Health++;
         }
 
-        public void LoseLifePoint(int loseLifeAmount = 1)
+        public void LoseLifePoint(int loseLifeAmount = 1) => LoseLifePoint(null, loseLifeAmount);
+        
+        
+        public void LoseLifePoint(Player responsible, int loseLifeAmount)
```
Comment (bashnia007) [inline Bang/Players/Player.cs:156]: **minor:** We almost everywhere call LoseLifePoint() method with loseLifeAmount parameter equals 1. Maybe we can set it as default parameter

File: Bang.Tests/PlayerEliminationSpecification.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+using Xunit;
+
+using static Bang.Game.GamePlayInitializer;
+
+namespace Bang.Tests
+{
+    public class PlayerEliminationSpecification
+    {
+        [Fact]
+        public void if_the_sheriff_eliminates_a_deputy_the_sheriff_must_discard_all_the_cards()
```
[the remark this answers is not in the document]
Comment (IvanovAndrew) [reply Bang.Tests/PlayerEliminationSpecification.cs]: Fixed

File: Bang/Game/PlayerEliminator.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            
+            if (victim.IsAlive)
+                throw new ArgumentException($"Player {victim.Name} is still alive!");
+
+            if (responsible != null && responsible != victim)
+            {
+                switch (victim.Role)
+                {
+                    // Any player eliminating an Outlaw must draw a reward of 3 cards from the deck.
+                    case Outlaw o:
```
[the remark this answers is not in the document]
Comment (IvanovAndrew) [reply Bang/Game/PlayerEliminator.cs]: Fixed

File: Bang/Game/PlayerEliminator.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                        responsible.TakeCards(3);
+                        break;
+
+                    // If the Sheriff eliminates a Deputy, the Sheriff must discard all the cards he has in hand and in play.
+                    case Deputy d:
```
[the remark this answers is not in the document]
Comment (IvanovAndrew) [reply Bang/Game/PlayerEliminator.cs]: Fixed

File: Bang/Game/PlayerEliminator.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                        if (responsible.Role is Sheriff s) responsible.DropAllCards();
```
[the remark this answers is not in the document]
Comment (IvanovAndrew) [reply Bang/Game/PlayerEliminator.cs]: Replaced into `is Sheriff`

File: Bang/Players/Player.cs (hunk 1 above)
Comment (IvanovAndrew) [reply Bang/Players/Player.cs]: Fixed

**Pull request merged**